swarm/storage/resource.go

```diff
@@ -0,0 +1,529 @@
+package storage
+
+import (
+	"crypto/ecdsa"
+	"encoding/binary"
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"time"
+
+	"golang.org/x/net/idna"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/contracts/ens"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+const (
+	SIGNATURE_LENGTH = 65
+	MAX_CHUNK_DATA   = 4096 - SIGNATURE_LENGTH
+	INDEX_SIZE       = 24
+)
+
+// Encapsulates an actual resource update. When synced it contains the most recent
+// version of the resource update data.
+type resource struct {
+	name       string
+	ensname    common.Hash
+	startblock uint64
+	lastblock  uint64
+	frequency  uint64
+	version    uint64
+	data       []byte
+	updated    time.Time
+}
```

Comment:
Is there a reason why multiword names do not follow camelcase here? startBlock, ensName, lastBlock

Reply:
fixed